meilisearch-types/src/keys.rs

```diff
@@ -179,142 +180,149 @@ fn parse_expiration_date(
+        ("settings.update", Self::SettingsUpdate),
+        ("stats.*", Self::StatsAll),
+        ("stats.get", Self::StatsGet),
+        ("metrics.*", Self::MetricsAll),
+        ("metrics.get", Self::MetricsGet),
+        ("dumps.*", Self::DumpsAll),
+        ("dumps.create", Self::DumpsCreate),
+        ("snapshots.*", Self::SnapshotsAll),
+        ("snapshots.create", Self::SnapshotsCreate),
+        ("version", Self::Version),
+        ("keys.create", Self::KeysAdd),
+        ("keys.get", Self::KeysGet),
+        ("keys.update", Self::KeysUpdate),
+        ("keys.delete", Self::KeysDelete),
+        ("experimental.get", Self::ExperimentalFeaturesGet),
+        ("experimental.update", Self::ExperimentalFeaturesUpdate),
+    ];
+
+    fn get_action(v: &str) -> Option<Action> {
+        Self::SERDE_MAP_ARR
+            .iter()
+            .find(|(serde_name, _)| &v == serde_name)
+            .map(|(_, action)| *action)
+    }
+
+    fn get_action_serde_name(v: &Action) -> &'static str {
+        Self::SERDE_MAP_ARR
+            .iter()
+            .find(|(_, action)| v == action)
+            .map(|(serde_name, _)| serde_name)
+            .expect("an action doesn't have a matching serialized value")
+    }
+
+    pub const fn from_repr(repr: u32) -> Option<Self> {
         use actions::*;
         match repr {
-            ALL => Some(Self::All),
             SEARCH => Some(Self::Search),
-            DOCUMENTS_ALL => Some(Self::DocumentsAll),
             DOCUMENTS_ADD => Some(Self::DocumentsAdd),
             DOCUMENTS_GET => Some(Self::DocumentsGet),
             DOCUMENTS_DELETE => Some(Self::DocumentsDelete),
-            INDEXES_ALL => Some(Self::IndexesAll),
+            DOCUMENTS_ALL => Some(Self::DocumentsAll),
             INDEXES_CREATE => Some(Self::IndexesAdd),
             INDEXES_GET => Some(Self::IndexesGet),
             INDEXES_UPDATE => Some(Self::IndexesUpdate),
             INDEXES_DELETE => Some(Self::IndexesDelete),
             INDEXES_SWAP => Some(Self::IndexesSwap),
-            TASKS_ALL => Some(Self::TasksAll),
+            INDEXES_ALL => Some(Self::IndexesAll),
             TASKS_CANCEL => Some(Self::TasksCancel),
             TASKS_DELETE => Some(Self::TasksDelete),
             TASKS_GET => Some(Self::TasksGet),
-            SETTINGS_ALL => Some(Self::SettingsAll),
+            TASKS_ALL => Some(Self::TasksAll),
             SETTINGS_GET => Some(Self::SettingsGet),
             SETTINGS_UPDATE => Some(Self::SettingsUpdate),
-            STATS_ALL => Some(Self::StatsAll),
+            SETTINGS_ALL => Some(Self::SettingsAll),
             STATS_GET => Some(Self::StatsGet),
-            METRICS_ALL => Some(Self::MetricsAll),
+            // @TODO: Issue: Since stats has only one element, all is the same as the one single element
+            //               so this will never match all, because it matches that one and only element first
+            STATS_ALL => Some(Self::StatsAll),
```

Comment:
I'm not sure if this will results in problems.

Reply:
Maybe you'll have to replace entirely this art 🤔

Reply:
It does, read more below.